OneCommunityGlobal/HighestGoodNetworkApp
# Kurtis - Fix date format inconsistencies in app

<div id="user-content-toc">
  <ul align="center">
    <summary><h1 style="display: inline-block">Fix date format inconsistencies in app</h1></summary>
     <summary><h1 style="display: inline-block">PR 1082</h1></summary>
     <summary><h2 style="display: inline-block">Finally finished!</h2></summary>

  </ul>
</div>
<img src="https://user-images.githubusercontent.com/73097560/115834477-dbab4500-a447-11eb-908a-139a6edaec5c.gif">

# Description
Inconsistency in the date format under weekly summary reports was noticed. Therefore, a fix was implemented as it was just a small formatting issue most likely. Luckily, the tz package pretty much takes care of it since it handles most if not all date formatting in the app. The date format in .tz.format ~~ was fixed to the desired format to address the issue.
~ in addition to that, I was told to fix all the instances of the date format in 'YYYY-MM-DD' and change it to the appropriate format 'MMM-DD-YY'


## Related PRS (if any):
Test with any backend. Personally, I used the current HGN rest dev backend
…

## Main changes explained:
- Change .format('YYYY-MMM-DD') to be   .format('MMM-DD-YY');
- enables date format consistency in weekly summary reports page
- Certain instances require configuring the Moment package to also be set to the "MMM-DD-YY"


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as an admin > verify weekly summary countdown timer has proper date format
4. go to reports > weekly summary reports > and double check the the date format is the same
5. go to view profile > manually pause profile to trigger an end date. Verify the dates in the correct format under the basic information tab.
6. go to view profile > select features badges > date should be in proper format such as "Jul-25-23" , must work on desktop and tablet res
7. go to view profile > look at the blue squares present and hover over them > date should be in proper format
8. go to timelog > current week timelog > verify date format in format such as "Jul-25-23"
9. <strong>(must be logged in as admin or higher)</strong>go to other links tab > usermanagement> pause someone, and verify the date format. be sure to unpause them if it's not your own account.
10. go to reports > teams > click any team to view created date format and verify in proper format. In addition, there is an invisible element with the class "update-date" you'll need to verify in the dom via inspecting on chrome. Search with the class and you'll find it.
11. dashboard > click any task in the "tasks" section to go to the individual task. verify start and end date format.
12. dashboard > other links > projects > wbs > click one of the wbs task names > hopefully you found one with a start and end date, so you can verify it.



## Screenshots or videos of changes:
inconsistent format as noted in photo:

![old date format pr1082](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/3fea653c-c4f1-4f42-ae09-64194007e594)


fixed:

![fix weekly summ date consistency per jae](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/6edde7d0-c101-409e-8d02-b8ecc41c613f)

badge format desktop res:
![pr1082 badge date format desktop](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/f8d4b837-3b5a-4b88-a24a-be79a37e3d68)
badge format tablet res:
![pr1082 badge date format tablet](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/e62ac4a1-1c2a-4b34-8204-0795107dedf3)
user report date format:
![pr1082 update user report date format on individual](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/6bc66557-2dcd-4e2f-b006-87554751974f)
team report date format:
![pr1082 fix format  team reports last updated date however is invisible](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/18dff3d4-ec6c-4941-9afb-9358a7b2ca74)
task date format:
![pr1082 task date format](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/29cf4314-4eac-4949-9a44-ed2154fad54a)
wbs date format:
![pr1082 wbs task date format start and end](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/29cdb2c3-43e0-4bf1-b74a-70632e8990c0)
weekly summary countdown timer format:
![pr1082 weekly summary due date countdown date format fix](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/ceecf7ae-6f42-46bd-9693-6b9db27b6cba)
blue square date format fix
![pr1082 fix date format from bluesquare description via custom function](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/19482363/b2dd3278-e76a-4c63-915b-0866202b3507)



notes:
Basic information tab in user profile cannot be changed due to a date format error that occurs on the backend. Normal use shouldn't interact with this area much, so low priority.
